syntax = "proto3";

option go_package = "github.com/e-conomic/hiring-assignments/machinelearningteam/summary-statistics-service/summary";

package summary_statistics;

service DocumentSummarizer {
  rpc SummarizeDocument (SummarizeDocumentRequest) returns (SummarizeDocumentReply) {}
}

message SummarizeDocumentRequest {
  Document document = 1;
  string params_for_aggregation = 2;
  string exclude = 3;
}

message SummarizeDocumentReply {
  bytes content = 1;
}

message Document {
  // Document content, represented as a stream of bytes.
  // Note: As with all `bytes` fields, protobuffers use a pure binary
  // representation, whereas JSON representations use base64.
  bytes content = 1;
  // If both content and source is present use content.
  DocumentSource source = 2;
}

message DocumentSource {
  // The URI of the source document. Can be either:
  //
  // 1. A publicly-accessible image HTTP/HTTPS URL. When fetching images from
  string http_uri = 1;
}
